### PR TITLE
  [symex] remove dead code in builtin_functions/ (codecov cleanup)

### PR DESCRIPTION
**Pull request by LukeW1999**
## Summary

Cleans up the dead-code subset of the codecov coverage gap in the new `src/goto-symex/builtin_functions/` split from #4166. Covers 4 of the 8 flagged files; the other 4 were analysed and intentionally left alone (see below).

**Net change**: 68 lines removed across 4 files, 0 behavioural changes.

## Per-file breakdown

| File | Δ | What was removed |
|---|---|---|
| `memory_ops.cpp` | -53 | `offset_simplifier` div(mul,c)/c branch; `guard.is_false()` early returns in `intrinsic_memcpy`/`intrinsic_memset`; symbolic-offset and inline div-simplifier blocks in `intrinsic_memset` (redundant with the `!is_constant_int2t` check below) |
| `io.cpp` | -6 | `symex_printf` `guard.is_false()` early-continue in the pointer-argument loop; collapsed an `if (spec != 'n') / else` pair with identical bodies into a single `actual_format_count++` |
| `object_size.cpp` | -4 | Collapsed `if (param_val < 0 \|\| param_val > 3) { type_value = 0; } else ...` into a positive-guard `if`; the `type_value = 0` assignment was redundant (variable zero-initialised at declaration) |
| `run_builtin.cpp` | -5 | Unreachable `else { log_error("Unknown overflow intrinsics"); abort(); }` in the overflow-intrinsic dispatcher — the outer `has_prefix` chain matches exactly the six prefixes that `is_mult`/`is_add`/`is_sub` exhaustively cover |

## Intentionally kept

The majority of codecov-flagged lines are **defensive fallbacks or rare live paths, not dead code**. Deleting them would introduce:

- **Infinite loops** — e.g. removing `i++` from format-string parser `while`-loops, or `alloc_obj = ...` in the `while (is_if2t(alloc_obj))` unwrap loops in `threads.cpp` / `python_builtins.cpp`
- **Crashes / UB** — removing `bump_call + return` pairs lets `to_constant_int2t` assert on non-constant input; removing exception handlers leaves `type_size` uninitialised for VLAs
- **Soundness bugs** — removing the constant-size `realloc` copy loop silently returns new memory without copying; removing empty-deref early-return makes `memcpy` a no-op instead of falling back to `__memcpy_impl`
- **Feature removal** — `threads.cpp` / `python_builtins.cpp` entries ARE the implementations of the thread and Python-operator intrinsics

Files skipped entirely: `memory_alloc.cpp`, `threads.cpp`, `python_builtins.cpp`, `va_arg.cpp`.

## Why codecov flagged them

These paths are uncovered because the regression suite doesn't exercise certain input shapes — fast-path fallbacks triggered by symbolic sizes and offsets, rare type combinations (pointer memset, bitfield structs, Optional-through-void-pointer), option-gated branches (`max-symbolic-realloc-copy`, `malloc-zero-is-null`, etc.), and thread/monitor intrinsics rarely called directly by concurrency tests. None of that is dead code.

## Follow-up

I will open a separate PR shortly to **add regression tests** targeting the live-but-untested paths inventoried while doing this cleanup (constant-size `realloc`, `memset` on bitfield/union/pointer, `__builtin_object_size(&arr[i])`, `hasattr`/`isinstance` corner cases, thread intrinsic direct invocation, etc.).

Assisted-by: Claude-Opus4.7

### CODE REVIEW

File: src/goto-symex/builtin_functions/run_builtin.cpp [hunk 1]
```diff
@@ -74,11 +74,6 @@ bool goto_symext::run_builtin(
     else if (is_sub)
       op = sub2tc(
         func_type.arguments[0], func_call.operands[0], func_call.operands[1]);
-    else
```
Comment (lucasccordeiro) [inline src/goto-symex/builtin_functions/run_builtin.cpp:77]: It is fine to keep this one.

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]

File: src/goto-symex/builtin_functions/run_builtin.cpp (hunk 1 above)
Comment (LukeW1999) [reply src/goto-symex/builtin_functions/run_builtin.cpp:77]: Added back.

**Review by lucasccordeiro — APPROVED** [review]
Thanks for submitting this PR, @LukeW1999.
